machine-learning/04-Model Evaluation.ipynb

```diff
@@ -4,25 +4,478 @@
+          "language": "csharp"
+        }
+      },
+      "source": [
+        "var trainTestData = mlContext.Data.TrainTestSplit(df,testFraction:0.2);\n",
+        "var validationTestData = mlContext.Data.TrainTestSplit(trainTestData.TestSet,testFraction:0.5);"
+      ],
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        }
+      },
+      "source": [
+        "var trainSet = trainTestData.TrainSet;\n",
+        "var validationSet = validationTestData.TrainSet;\n",
+        "var testSet = validationTestData.TestSet;"
+      ],
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Create training pipeline\n",
+        "\n",
+        "For ths dataset, the following transforms are applie:\n",
+        "\n",
+        "- OneHotEncoding to convert categorical values into numerical values\n",
+        "- ReplaceMIssingValues which as the name suggests is to replace any missing values.\n",
+        "- Concatenate takes all of the features and creates a feature vector\n",
+        "\n",
+        "AutoML is used to define a regression experiment using the `fare_amount` column as the column to predict or label column. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        }
+      },
+      "source": [
+        "var pipeline = \n",
+        "\tmlContext.Transforms.Categorical.OneHotEncoding(new[] { new InputOutputColumnPair(@\"vendor_id\", @\"vendor_id\"), new InputOutputColumnPair(@\"payment_type\", @\"payment_type\")},outputKind: OutputKind.Binary)\n",
+        "\t\t.Append(mlContext.Transforms.ReplaceMissingValues(new[] { new InputOutputColumnPair(@\"rate_code\", @\"rate_code\"), new InputOutputColumnPair(@\"passenger_count\", @\"passenger_count\"), new InputOutputColumnPair(@\"trip_time_in_secs\", @\"trip_time_in_secs\"), new InputOutputColumnPair(@\"trip_distance\", @\"trip_distance\") }))\n",
+        "        .Append(mlContext.Transforms.Concatenate(@\"Features\", new[] { @\"vendor_id\", @\"payment_type\", @\"rate_code\", @\"passenger_count\", @\"trip_time_in_secs\", @\"trip_distance\" }))\n",
+        "        .Append(mlContext.Auto().Regression(labelColumnName: \"fare_amount\"));"
+      ],
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Configure experiment\n",
+        "\n",
+        "Use AutoML to configure our experiment to train for 60 seconds using the pipeline you've just defined. \n",
+        "\n",
+        "By default, AutoML evaluates the models it trains using the evaluation metric you want to optimize. In this case it's R-Squared which is calculated by comaring the actual value `fare_amount` against the predicted value `Score`. "

```

Comment:
I was interested in hearing more examples here on other metrics I may want to optimize on. Not necessary if you can't think of another good example here.

Reply:
Resolved.